### PR TITLE
Add new option to spark-run to choose which cluster manager to use

### PR DESCRIPTION
**Pull request by gcollura**
## What changed
- [x] add option `--cluster-manager mesos|kubernetes` to pick which CM to use
- [x] correctly load ALL containers in the Spark driver to match the behavior with Mesos (https://github.com/Yelp/service_configuration_lib/pull/64)
- [x] update paasta tron to not mount the same volume (`/etc/pki/spark`) twice  

## Testing
Manual
```
./paasta_tools/cli/cli.py spark-run --cluster-manager kubernetes --cluster pnw-devc --cmd ' spark-submit /code/virtualenv_run/bin/spark_etl_runner.py --feature-config /code/batch/xxx/job_configs/xxx.yaml --env-config-path /nail/srv/configs/xxx.yaml --team-name xxx --notify-email y@elp.com --start-date 2021-08-15 --end-date 2021-08-15 --publish-path s3a://yelp-xxx-west-2/x/ -v ' --spark-args ' spark.cores.max=60 spark.executor.memory=13g spark.executor.cores=2 spark.driver.memory=13g spark.driver.cores=2 spark.jars.ivy=/tmp/.ivy spark.executor.instances=30' --service xxx
```
Succeeds: https://fluffy.yelpcorp.com/i/LfQ0XQ47l5s0jCF1pbnBS289ddH855mx.html

### CODE REVIEW

File: paasta_tools/cli/cmds/spark_run.py
Comment (88manpreet) [inline paasta_tools/cli/cmds/spark_run.py:565]: How are spark.kubernetes.executor.volumes.host* being set in spark_conf at first place? Through the env variables populated by the other review?

**Comment by 88manpreet** [comment]
lgtm
I guess the plan is to add the test results before merging this? 
Please also add @vkhromov  and Li Rong for any reviews or visibility. 
Getting it reviewed by someone from #compute-infra for paasta and service_configuration_lib changes would be required.

File: paasta_tools/cli/cmds/spark_run.py
Comment (gcollura) [reply paasta_tools/cli/cmds/spark_run.py:565]: Missed your comment, yes they're set in service-configuration-lib

File: tests/cli/test_cmds_spark_run.py
Comment (88manpreet) [inline tests/cli/test_cmds_spark_run.py:396]: Nit: For consistency, maybe use CLUSTER_MANAGER_MESOS? 
Same as everywhere else.

**Comment by 88manpreet** [comment]
Is the idea to test it after deploying?

**Comment by 88manpreet** [comment]
I think this would also require at least unit test for kubernetes cluster-manager?

File: paasta_tools/cli/cmds/spark_run.py
Comment (88manpreet) [inline paasta_tools/cli/cmds/spark_run.py:555]: Can we add an info log-line somewhere, to display which cluster-manager is being used?

**Review by 88manpreet — APPROVED** [review]

**Comment by 88manpreet** [comment]
lgtm

**Review by 88manpreet — APPROVED** [review]
lgtm once @chelspea approves it

**Pull request merged**
